### DISCUSSION TITLE
400 Bad Request Error when reconfiguring existing areas after updating to v4.0.0

### DISCUSSION THREAD

**Discussion by mihsu81** [jseidl/magic-areas#360] [Q&A] [answered]
Getting an error when reconfiguring existing areas after updating to v4.0.0.
It does seem to try loading the config flow for a split second before the error appears.

No errors/warnings related to Magic Areas are present in the logs.
I'm attaching debug logs.
[home-assistant_magic_areas_2024-06-07T05-52-43.450Z.log](https://github.com/user-attachments/files/15710836/home-assistant_magic_areas_2024-06-07T05-52-43.450Z.log)

I did clear cookies and the cache of Home Assistant Android app, but it didn't help.

There are a few exceptions though which can be reconfigured without throwing an error: Exterior, Global, Interior, Office, which only have the Area State sensor. 

```
Error
Config flow could not be loaded: 400: Bad Request
```
![image](https://github.com/jseidl/hass-magic_areas/assets/40715497/5444a57d-ce61-4969-9c99-2c5083847854)

**Comment by pascalmtts** [comment]
I had the same problem, had to reconfigure all areas

**Reply by mihsu81** [reply]
That's something I really hope to avoid if at all possible. @jseidl didn't mention any reconfiguration being needed, so I hope it's just an annoying bug.

**Comment by jseidl** [comment, marked as answer]
Hi! Yes sorry about the misunderstanding. Every major release (when the first digit of the version goes up) is potentially breaking. 

I've mentioned on the end of the release notes:
> If anything looks broken, try first deleting and re-creating the Magic Area. If it is still broken, cut me an [issue](https://github.com/jseidl/hass-magic_areas/issues) or come chat on Discord!

Although I've tested a lot during the beta, there are a few odd cases where you might have to delete and re-create some entries. The logs aren't helping much so I wasn't able to pin down what it is but since the config schema changed it might be that.

**Reply by d4g** [reply]
To be totally honest, this is not good in any way. I understand that maintaining a project is a big task and takes lots of effort and time, but not migrating pre-existing settings is just bad. I now need to recreate everything from the scratch and this will probably be the case for all your users.

**Reply by jseidl** [reply]
I understand and agree 100%. I myself had to do it as well for my 30 areas. I'm still learning how config migration works and hope to get better if I need to do it again.

Appreciate everyone's patience on this.

**Reply by mihsu81** [reply]
In the end I reconfigured the areas manually based on _**homeassistant/.storage/core.config_entries**_ to make sure I keep the same exceptions and settings. Time consuming but enough for me.

Thank you for your great work on this integration. 😊
